### PR TITLE
Remove support for Python 2.6

### PR DESCRIPTION
**Pull request by alalazo**
Modifications:
- [x] Removed `centos:6` unit test, adjusted vermin checks
- [x] Removed backport of `collections.OrderedDict`
- [x] Removed backport of `functools.total_ordering`
- [x] Removed Python 2.6 specific skip markers in unit tests
- [x] Fixed a few minor Python 2.6 related TODOs in code

Updating the vendored dependencies will be done in separate PRs

### CODE REVIEW

**Comment by adamjstewart** [comment]
We can also remove argparse

**Comment by alalazo** [comment]
I'm working through the dependencies. I think `argparse` needs to stay for Python 3.X compatibility.

**Comment by alalazo** [comment]
Since updating vendored dependencies is resulting in big changesets that might obfuscate the changes done in Core Spack, I'll submit separate PRs for those.

**Comment by adamjstewart** [comment]
Can you update the Getting Started docs too?

**Comment by alalazo** [comment]
@adamjstewart Should be done

File: lib/spack/docs/contribution_guide.rst [hunk 1]
```diff
@@ -71,7 +71,7 @@ locally to speed up the review process.
    new release that is causing problems. If this is the case, please file an issue.
 
 
-We currently test against Python 2.6, 2.7, and 3.5-3.7 on both macOS and Linux and
+We currently test against Python 2.7 and 3.5-3.7 on both macOS and Linux and
```
Comment (adamjstewart) [inline lib/spack/docs/contribution_guide.rst]:
```suggestion
We currently test against Python 2.7 and 3.5-3.9 on both macOS and Linux and
```

File: lib/spack/external/__init__.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
 --------
 
 * Homepage: https://pypi.python.org/pypi/argparse
-* Usage: We include our own version to be Python 2.6 compatible.
+* Usage: We include our own version to be Python 3.X compatible.
```
Comment (adamjstewart) [inline lib/spack/external/__init__.py:20]: Should this be 2.X compatible? I'm surprised we still need to vendor this even in Python 2.7.

File: lib/spack/spack/cmd/license.py [hunk 3]
```diff
@@ -46,9 +46,8 @@
     r'^lib/spack/docs/.*\.py$',
     r'^lib/spack/docs/spack.yaml$',
 
-    # 2 files in external
+    # 1 files in external
```
Comment (adamjstewart) [inline lib/spack/spack/cmd/license.py]:
```suggestion
    # 1 file in external
```

File: bin/spack [hunk 4]
```diff
@@ -54,8 +54,6 @@ spack_external_libs = os.path.join(spack_lib_path, "external")
 
 if sys.version_info[:2] <= (2, 7):
     sys.path.insert(0, os.path.join(spack_external_libs, "py2"))
-if sys.version_info[:2] == (2, 6):
-    sys.path.insert(0, os.path.join(spack_external_libs, "py26"))
```
Comment (tgamblin) [inline bin/spack:58]: The condition and message above (on lines 36-42) needs to be changed as well.

File: lib/spack/external/__init__.py (hunk 2 above)
Comment (tgamblin) [reply lib/spack/external/__init__.py:20]: 2.7 unfortunately doesn't have aliases.

File: lib/spack/external/__init__.py [hunk 5]
```diff
@@ -73,14 +64,6 @@
 * Usage: Implements a XML/HTML/XHTML Markup safe string for Python.
 * Version: 1.0
 
-orderddict
-----------
-
-* Homepage: https://pypi.org/project/ordereddict/
-* Usage: A drop-in substitute for Py2.7's new collections.OrderedDict
```
Comment (tgamblin) [inline lib/spack/external/__init__.py:80]: "new". lol.

**Review by tgamblin — CHANGES_REQUESTED** [review]
One minor request in `bin/spack`

File: lib/spack/external/__init__.py (hunk 2 above)
Comment (alalazo) [reply lib/spack/external/__init__.py:20]: @adamjstewart See for instance:

https://github.com/spack/spack/blob/feb66cba011446da25b90982de7a29d869d9c53e/lib/spack/spack/cmd/compiler.py#L35-L37

**Comment by alalazo** [comment]
@tgamblin @adamjstewart Ready for another review

**Review by adamjstewart — APPROVED** [review]

**Review by tgamblin — APPROVED** [review]

**Pull request merged**